### PR TITLE
refactor: node install buildin cache

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

- Support cache of dependencies in setup-node.

## Context:

- Fixes: #114

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
- move checkout before `setup-node`
- remove other cache stuff

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,35 +20,23 @@ jobs:
     timeout-minutes: 30
```
Comment (HonkingGoose) [inline .github/workflows/build.yml:20]: I notice we have a env variable called `YARN_CACHE_FOLDER` on line 12 of this file, that points to `.cache/yarn`. This line should be deleted, as I don't see us _using_ the variable in the CI script anymore.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (viceice) [reply .github/workflows/build.yml:20]: Yes, please remove those yarn env values

**Review by viceice — COMMENTED** [review]
See comments

**Review by viceice — APPROVED** [review]

**Pull request merged**